controlplane/kubeadm/controllers/kubeadm_control_plane_controller.go

```diff
@@ -324,7 +331,60 @@ func (r *KubeadmControlPlaneReconciler) updateStatus(ctx context.Context, kcp *c
 	return nil
 }
 
-func (r *KubeadmControlPlaneReconciler) upgradeControlPlane(ctx context.Context, cluster *clusterv1.Cluster, kcp *controlplanev1.KubeadmControlPlane, ownedMachines internal.FilterableMachineCollection, requireUpgrade internal.FilterableMachineCollection) (ctrl.Result, error) {
+// updateCoreDNS updates the workload cluster's kubeadm config, coredns
+// corefile and coredns deployment to match the appropriate imageTag specified
+// in the kcp DNS.
+func (r *KubeadmControlPlaneReconciler) updateCoreDNS(
+	ctx context.Context,
+	cluster *clusterv1.Cluster,
+	kcp *controlplanev1.KubeadmControlPlane,
+) error {
```

Comment:
Not blocking: we usually don't use this style for methods/functions, rather if a function is taking too many parameters we often switch to an `Input` struct instead